Modules/System/WebDAV/src/WebDAVClientImpl.Codeunit.al

```diff
@@ -0,0 +1,192 @@
+    end;
+
+    procedure GetFilesAndCollections(var WebDAVContent: Record "WebDAV Content"; Recursive: Boolean): Boolean
+    var
+        WebDAVContentParser: Codeunit "WebDAV Content Parser";
+        ResponseContent: Text;
+    begin
+        ResponseContent := GetPropfindResponse(Recursive);
+
+        WebDAVContentParser.Initialize(Uri, false, false);
+        WebDAVContentParser.Parse(ResponseContent, WebDAVContent);
+
+        exit(true);
+    end;
+
+    procedure GetCollections(var WebDAVContent: Record "WebDAV Content"; Recursive: Boolean): Boolean
+    var
+        WebDAVContentParser: Codeunit "WebDAV Content Parser";
+        ResponseContent: Text;
+    begin
+        ResponseContent := GetPropfindResponse(Recursive);
+
+        WebDAVContentParser.Initialize(Uri, false, true);
+        WebDAVContentParser.Parse(ResponseContent, WebDAVContent);
+        exit(true);
+    end;
+
+    procedure GetFiles(var WebDAVContent: Record "WebDAV Content"; Recursive: Boolean): Boolean
+    var
+        WebDAVContentParser: Codeunit "WebDAV Content Parser";
+        ResponseContent: Text;
+    begin
+        ResponseContent := GetPropfindResponse(Recursive);
+
+        WebDAVContentParser.Initialize(Uri, true, false);
+        WebDAVContentParser.Parse(ResponseContent, WebDAVContent);
+
+        exit(true);
+    end;
+
+    local procedure GetPropfindResponse(Recursive: Boolean) ResponseContent: Text;
+    var
+        WebDAVRequestHelper: Codeunit "WebDAV Request Helper";
+    begin
+        WebDAVRequestHelper.SetAuthorization(Authorization);
+        WebDAVOperationResponse := WebDAVRequestHelper.Propfind(Uri, Recursive);
+
+        // TODO Success??
+        if not WebDAVOperationResponse.GetDiagnostics().IsSuccessStatusCode() then
+            Error('%1: %2', WebDAVOperationResponse.GetDiagnostics().GetHttpStatusCode(), WebDAVOperationResponse.GetDiagnostics().GetResponseReasonPhrase());
+
+        WebDAVOperationResponse.TryGetResponseAsText(ResponseContent);
+    end;
+
+    procedure GetFileContent(var ResponseInStream: InStream): Boolean
+    var
+        WebDAVRequestHelper: Codeunit "WebDAV Request Helper";
+    begin
+        WebDAVRequestHelper.SetAuthorization(Authorization);
+        WebDAVOperationResponse := WebDAVRequestHelper.Get(Uri);
+
+        if not WebDAVOperationResponse.GetDiagnostics().IsSuccessStatusCode() then
+            exit(false);
+
+        if WebDAVOperationResponse.TryGetResponseAsStream(ResponseInStream) then
+            exit(true);
+    end;
+
+    procedure GetFileContentAsText(var ResponseText: Text): Boolean
+    var
+        ResponseInStream: InStream;
+    begin
+        GetFileContent(ResponseInStream);
+        ResponseInStream.Read(ResponseText);
+        exit(true);
+    end;
+
+    procedure GetDiagnostics(): Interface "HTTP Diagnostics"
+    begin
+        exit(WebDAVOperationResponse.GetDiagnostics());
+    end;
+
+    [TryFunction]
+    procedure TryGetResponseAsText(var Response: Text)
```

Comment:
Again these two functions should just return the response directly instead of using a by-var variable.